server/src/main/java/za/dots/db/DatasourceConnection.java

```diff
@@ -0,0 +1,21 @@
+package za.dots.db;
+
+import java.sql.*;
+
+public class DatasourceConnection {
+
+    Connection connection;
+
+    private String cleanInputValue(String string) {
+        return string.replaceAll("[^a-zA-Z\\d]", "");
+    }
+
+    public DatasourceConnection() throws SQLException {
+        this.connection = DriverManager.getConnection("jdbc:sqlserver://"
```

Comment:
Isn't it a better practice to have a function to connect to the database (which returns Connection) and a function to close the connection (which takes in the connection)?

Reply:
because opening the connection in the constructor means the connection is open as soon an object is created of this class, and this object can be created in the constructor of the controller

Reply:
I was thinking we use this object for DB operations, that's why I connect in constructor (if it fails then no db operations can happen) will add a method to close connection